### PR TITLE
Manage referenced_by for shared files/folders

### PR DESCRIPTION
**Pull request by nono**

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -35,6 +35,7 @@
 * `/jobs` - [Jobs](jobs.md)
   * [Konnectors](konnectors.md)
   * [Workers](workers.md)
+- `/move` - [Move, export and import an instance](move.md)
```
Comment (Gara64) [inline docs/README.md]: Should it be part of the PR? :)

File: docs/toc.yml [hunk 2]
```diff
@@ -27,6 +27,7 @@
   - "/jobs Jobs": ./jobs.md
   - "Konnectors": ./konnectors.md
   - "Workers": ./workers.md
+  - "/move - Move, export and import an instance": ./move.md
```
Comment (Gara64) [inline docs/toc.yml]: Same question

File: pkg/sharing/rule.go [hunk 3]
```diff
@@ -200,6 +201,39 @@ func (s *Sharing) FirstFilesRule() *Rule {
...
+		allFound := true
+		for _, ref := range file.ReferencedBy {
+			v := ref.Type + "/" + ref.ID
+			found := false
+			for _, val := range rule.Values {
+				if val == v {
+					found = true
+					break
+				}
+			}
```
Comment (Gara64) [inline pkg/sharing/rule.go]: You could use `rule.hasReferencedBy(ref)`

File: docs/README.md (hunk 1 above)
Comment (nono) [reply docs/README.md]: No, it's not related. It was just that I was working on fixing a typo on the sharings docs, and I saw this entry was missing. I added too in a commit about documentation.

File: pkg/sharing/rule.go [hunk 4]
```diff
@@ -200,6 +201,39 @@ func (s *Sharing) FirstFilesRule() *Rule {
...
+			if !found {
+				allFound = false
+				break
+			}
+		}
+		if allFound {
+			return &s.Rules[i]
+		}
```
[the remark this answers is not in the document]
Comment (nono) [reply pkg/sharing/rule.go]: When a cozy send a new file to another cozy, it is expected that it sends only the relevant references (ie the references tied to a sharing rule). In practice, having several relevant references on the same file will be an exception. Here, it's more a sanity check that the received file has no references that the sender has forgot to remove (ie a reference to a not shared document).

**Pull request merged**